src/features/toilet/Toilet.js

```diff
@@ -1,22 +1,239 @@
+import squaredSOS from "../../assets/icon-squaredsos.svg";
+import viewFinder from "../../assets/icon-viewfinder.svg";
+import waitIcon from "../../assets/icon-wait-fluid.png";
+import ButtonDefault from "../../common/components/buttons/ButtonDefault";
+import ButtonFluid from "../../common/components/buttons/ButtonFluid";
 import HeaderSub from "../../common/components/headers/HeaderSub";
+import ListDefault from "../../common/components/lists/ListDefault";
+import ReviewCard from "../../common/components/reviewCard/ReviewCard";
+import StarContainer from "../../common/components/starContainer/StarContainer";
+import Title from "../../common/components/Title";
+import { updateToiletInfo } from "./toiletSlice";
 
 const StyledToilet = styled.div`
   width: 100%;
   display: flex;
   flex-direction: column;
   background-color: black;
+  color: white;
+
+  .titleContainer {
+    display: flex;
+    align-items: center;
+  }
+
+  .rankContainer {
+    display: flex;
+    align-items: center;
+    justify-content: space-evenly;
+  }
+
+  .toiletPaperContainer {
+    width: 100%;
+    display: flex;
+    align-items: center;
+  }
+
+  .lastToiletPaterProvideTime {
+    width: 100%;
+    font-size: small;
+    color: gray;
+    margin-right: 1rem;
+  }
 `;
 
 function Toilet() {
-  // const navigate = useNavigate();
+  const { toilet_id } = useParams();
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const location = useLocation();
+
+  const {
+    showToiletPath,
+    toiletName,
+    roadNameAddress,
+    inUnisexToilet,
+    menToiletBowlNumber,
+    menHandicapToiletBowlNumber,
+    menChildrenToiletBowlNumber,
+    ladiesToiletBowlNumber,
+    ladiesHandicapToiletBowlNumber,
+    ladiesChildrenToiletBowlNumber,
+    openTime,
+    latestToiletPaperInfo,
+    isSOS,
+    chatRoomList,
+  } = location.state;
+
+  const userClickedSOSButton = useSelector(
+    (state) => state.currnetToiletInfo.byIds[toilet_id]?.userClickedSOSButton
+  );
+  const isSOSCurrnetToilet = useSelector(
+    (state) => state.currnetToiletInfo.byIds[toilet_id]?.isSOSCurrnetToilet
+  );
+
+  const [reviews, setReviews] = useState([]);
+  const [avgRating, setAvgRating] = useState(0);
+  const [userSOSButton, setUserSOSButton] = useState(userClickedSOSButton);
+
+  useEffect(() => {
+    dispatch(updateToiletInfo({ toilet_id, isSOS, userSOSButton }));
+  }, [userSOSButton]);
+
+  useEffect(() => {
+    async function getReviews() {
+      try {
+        const { data } = await axios.get(`/toilets/review/${toilet_id}`);
+        setReviews(data.reviewList);
+      } catch (error) {
+        // 추후 에러처리 필요.
+        console.log(error);
+      }
+    }
+
+    getReviews();
+  }, []);
+
+  useEffect(() => {
+    function getAvgRating() {
+      if (!reviews.length) {
+        return 0;
+      }
+
+      let totalRating = 0;
+      // eslint-disable-next-line no-return-assign
+      reviews.forEach((review) => (totalRating += review.rating));
```

Comment:
여기서는 콜백 함수로 리턴해주는 값이 없어도 괜찮을것 같은데요. eslint를 끄시기 보다는 아래처럼 리팩토링 해보는건 어떨까요?

```suggestion
      reviews.forEach((review) => {totalRating += review.rating});
```

Reply:
수정하겠습니다!